linkerd/proxy/http/src/strip_header.rs

```diff
@@ -141,16 +148,16 @@ pub mod response {
 
     impl<F, H, B> Future for ResponseFuture<F, H>
     where
-        F: Future<Item = http::Response<B>>,
+        F: TryFuture<Ok = http::Response<B>>,
         H: AsHeaderName + Clone,
     {
-        type Item = F::Item;
-        type Error = F::Error;
+        type Output = Result<F::Ok, F::Error>;
 
-        fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-            let mut res = try_ready!(self.inner.poll());
-            res.headers_mut().remove(self.header.clone());
-            Ok(res.into())
+        fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+            let this = self.project();
+            let mut res = ready!(this.inner.try_poll(cx))?;
+            res.headers_mut().remove(this.header.clone());
+            Poll::Ready(Ok(res.into()))
```

Comment:
The `Into` conversion here was to turn the inner value into an `Async::Ready`; it's no longer necessary and we should remove it
```suggestion
            Poll::Ready(Ok(res))
```